### ISSUE TITLE
Disappearing Particles

### ISSUE THREAD

**Issue by laumeyertum** [AutoPas/AutoPas#692]
**Describe the bug**
Prticles get lost during the simulation of the falling Drop example with md-flexible.

**To Reproduce**
1 Compile md-flexible
2 Run md-flexible with the fallingDrop.yaml file
3 Check output for number of particles at start and end

**Expected behavior**
Consistent number of particles at beginning and end of the simulation.

**Output**
```bash
leon@DESKTOP-B29PTMN:~/AutoPas/build-clang/examples/md-flexible(master⚡) » ./md-flexible --yaml-filename fallingDrop.yaml
container                        :  [LinkedCells, VerletClusterLists, VerletLists, VerletListsCells]
verlet-rebuild-frequency         :  10
verlet-skin-radius               :  1
verlet-cluster-size              :  4
selector-strategy                :  Fastest-Absolute-Value
data-layout                      :  [AoS, SoA]
traversal                        :  [lc_c01, lc_c08, lc_c18, lc_sliced_c02, vcl_c01_balanced, vcl_c06, vcl_cluster_iteration, vl_list_iteration, vlc_c01, vlc_c18, vlc_sliced_c02]
tuning-strategy                  :  full-Search
mpi-strategy                     :  no-mpi
tuning-interval                  :  2500
tuning-samples                   :  3
tuning-max-evidence              :  10
functor                          :  Lennard-Jones (12-6)
newton3                          :  [disabled, enabled]
cutoff                           :  3
box-min                          :  [0, 0, 0]
box-max                          :  [49.5612, 29.5612, 37.296]
cell-size                        :  [1]
deltaT                           :  0.0005
iterations                       :  15000
boundary-type                    : [reflective, reflective, reflective]
Objects:
  Sphere:
    0:
      center                     :  [18, 15, 30]
      radius                     :  6
      particle-spacing           :  1.12246
      velocity                   :  [0, 0, 0]
      particle-type              :  1
      particle-epsilon           :  1
      particle-sigma             :  1
      particle-mass              :  1
globalForce                      :  [0, 0, -12]
vtk-filename                     :  fallingDrop
vtk-write-frequency              :  1000
log-level                        :  2
no-flops                         :  0
no-end-config                    :  1
no-progress-bar                  :  0
load-balancer                    :  InvertedPressure
load-balancing-interval          :  100
subdivide-dimension              :  [true, true, true]


Using 1 Threads
MPI is disabled.
Total number of particles at the initialization: 15092

Total number of particles at the end of Simulation: 13869
Owned: 13869
Halo : 0
```

**Build environment (please complete the following information):**
* OS: WSL2 Windows 10 running Ubuntu 20.04 | Lrz cluster
* Compiler: clang gcc9 (with and without mpi) intel
* Build System: make cmake

**Additional context**
Particles are being dropped somewhat consistently one by one. I cant detect a specific configuration that causes this behavior. Also the amount of disappearing particles varies each run.

**Comment by FG-TUM** [comment]
I just noticed VCL have the same `_towerSideLength` in each dimension. We should revisit this, maybe it has something to do with that?

**Comment by FG-TUM** [comment]
`VCL::updateHaloParticle()` might have ignored particles in the buffers. This detail should be fixed in #712. Need to check whether this affects this issue.

**Comment by dmartin13** [comment]
I spent some time to verify if this bug has been fixed. TL;TR: The bug is fixed.

Basically, I just took samples of merge commits to track down the solution to the problem.
The problem was not with Verlet Cluster Lists, as the bug also occurred with other containers. The same bug was basically mentioned in issue #713 , but sanity checks were already introduced here (I think sometime in November), so md-flex aborted when particles were lost. The bug was finally solved in PR #715. So the issue was not directly related to AutoPas, but to md-flex.

So I would suggest to close this issue.
